### PR TITLE
[aws][chore] Enable sandbox testing not maintained by git

### PR DESCRIPTION
**Pull request by anjafr**
# Description

Provides the `.sandbox/` directory, the contents of which are being ignore by git and thus can be used for experimental sandbox-y files that should not be tracked.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,5 @@ resotocore/tools/Assets
 out/
 *.iml
 **/.hypothesis/
+plugins/aws/test/resources/*.ipynb
```
Comment (aquamatthias) [inline .gitignore:35]: I guess this is might be helpful not only in the context of AWS - maybe we use a directory on project root level?

File: plugins/aws/resoto_plugin_aws/aws_client.py [hunk 2]
```diff
@@ -41,6 +41,9 @@ def __init__(
         self.role = role
         self.profile = profile
         self.region = region
+        self.AWS_ACCESS_KEY_ID = None
+        self.AWS_SECRET_ACCESS_KEY = None
+        self.AWS_SESSION_TOKEN = None
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/aws_client.py:46]: I don't think this is a good idea. Is there a better way to achieve this without changing production code?

**Review by aquamatthias — CHANGES_REQUESTED** [review]

File: plugins/aws/resoto_plugin_aws/aws_client.py (hunk 2 above)
Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/aws_client.py]: @aquamatthias all of this is removed, credentials are now being used via the corresponding credentials file (or other mechanisms that I don't know about )

File: .gitignore [hunk 3]
```diff
@@ -32,3 +32,6 @@ resotocore/tools/Assets
 out/
 *.iml
 **/.hypothesis/
+plugins/api_sandboxes/*.ipynb
+!plugins/api_sandboxes/*.example.ipynb
```
Comment (aquamatthias) [inline .gitignore:36]: Let's remove this line and the corresponding example file.
Create a `README.md` file that explains the reason for this folder, ala:

```
All files placed in this directory are ignored by git.
You can use this directory to put files and directories for sandbox testing.
```

File: .gitignore [hunk 4]
```diff
@@ -32,3 +32,6 @@ resotocore/tools/Assets
 out/
 *.iml
 **/.hypothesis/
+plugins/api_sandboxes/*.ipynb
```
Comment (aquamatthias) [inline .gitignore:35]: Why plugins?
I would opt for `.sandbox/` - the dot for hiding this directory and everything in it (no matter which extension).

**Review by aquamatthias — COMMENTED** [review]
Thanks!

To make this use case available without being prominent, I would opt for a dot directory on root level.

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Review by TheCatLady — APPROVED** [review]

**Pull request merged**